javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/instrumentation/indy/ClassCopySource.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.tooling.instrumentation.indy;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import net.bytebuddy.utility.StreamDrainer;
+
+/**
+ * Provides the bytecode and the original resource URL for loaded and not-yet loaded classes. The
+ * implementation is based on {@link net.bytebuddy.dynamic.ClassFileLocator.ForClassLoader}, with
+ * the difference that it preserves the original classfile resource URL.
+ */
+public abstract class ClassCopySource {
```

Comment:
This looks like it could be an interface instead of an abstract class... Consider changing.

Reply:
I acknowledge that @trask said below that the entire module is considered internal, but still think it's a good practice to at least hide this class from other potential implementers. If we don't, someone somewhere will try and inherit from this. 🙃 

Probably could make it package access and hide the constructor (private might work with inner classes, right?).